### ISSUE TITLE
cancer_classifier.py error

### ISSUE THREAD

**Issue by godsarmy** [tobegit3hub/tensorflow_template_application#21]
After fixing issue of #15, i met more complicated issue in `assert_is_compatible_with`

```
Traceback (most recent call last):
  File "./cancer_classifier.py", line 241, in <module>
    tf.app.run()
  File "/Library/Python/2.7/site-packages/tensorflow/python/platform/app.py", line 48, in run
    _sys.exit(main(_sys.argv[:1] + flags_passthrough))
  File "./cancer_classifier.py", line 174, in main
    concated = tf.concat(1, [indices, sparse_labels])
  File "/Library/Python/2.7/site-packages/tensorflow/python/ops/array_ops.py", line 1032, in concat
    ).assert_is_compatible_with(tensor_shape.scalar())
  File "/Library/Python/2.7/site-packages/tensorflow/python/framework/tensor_shape.py", line 735, in assert_is_compatible_with
    raise ValueError("Shapes %s and %s are incompatible" % (self, other))
ValueError: Shapes (2, 1024, 1) and () are incompatible
```

my tensorflow version:
```
python -c 'import tensorflow; print tensorflow.__version__'
1.1.0-rc0
```

**Comment by tobegit3hub** [comment]
Thanks for reporting.

I don't have `TensorFlow 1.1.0-rc0` but I have tested and succeed in `TensorFlow 1.0.1`.

I will test once the `1.1.0` is officially released.

**Comment by godsarmy** [comment]
Thanks.

 I think tensorflow 1.1 has some of its API changed

**Comment by qpanpony** [comment]
I met the same issue as  @godsarmy.
I also tested in both Tensorflow 1.1.0 and a degraded 1.0.1, but no success.

**Comment by tobegit3hub** [comment]
I have tested the code with the latest stable `TensorFlow 1.1.0` and eveything should work.

@godsarmy @qpanpony What's your version of TensorFlow and operation system? Please pull the code of `deep_recommend_system` and test again.

```
➜ python -c "import tensorflow; print(tensorflow.__version__)"
1.1.0
```

**Comment by qpanpony** [comment]
@tobegit3hub 
I tested it in docker image pulled from gcr.io/tensorflow/tensorflow:latest. Yes, the version of tensorflow in the container from this image is 1.1.0. 
Actually I tried it in a 2-node kubernetes cluster. I started ps (parameter server) node successfully, but when I tried to start worker node, it failed. 

Additional MSG: the operation system is Ubuntu 16.04.2 LTS.

**Comment by tobegit3hub** [comment]
The code of distributed training is kind of out-of-date. I will update for `TensorFlow 1.0.0+` APIs soon.

Thanks for repoerting @qpanpony .

**Comment by tobegit3hub** [comment]
I have fixed that and test with `TensorFlow 1.1.0`.

You can follow the document to run the latest `dense_classifier.py` in https://github.com/tobegit3hub/deep_recommend_system/tree/master/distributed .
